File: apps/api/app/repositories/mock_search_repository.py
from app.repositories.mock_data import get_search_suggestions
from app.repositories.search_repository import SearchRepository
from app.schemas.search import SearchSuggestion


class MockSearchRepository(SearchRepository):
    def search_stocks(self, query: str) -> list[SearchSuggestion]:
        return get_search_suggestions(query)

File: apps/web/repositories/mock/mock-search-repository.ts
import type { SearchRepository } from "@/repositories/search-repository";
import { getMockSearchSuggestions } from "@/repositories/mock/mock-source";
import type { SearchSuggestion } from "@/types/search";

export class MockSearchRepository implements SearchRepository {
  async searchStocks(query: string): Promise<SearchSuggestion[]> {
    return getMockSearchSuggestions(query);
  }
}

